### PR TITLE
Feat/pre commit linting hook & Monitoring widgets

### PR DESCRIPTION
**Pull request by Zequez**
Added pre-commit linting hook; should run Eslint on all staged files before committing which should also fix and format with Prettier.

- `husky` to `package.json`
- `lint-staged` to `frontend/package.json`

It also [makes TypeScript checks with the use of eslint-plugin-tsc as demonstrated on this issue.](https://github.com/okonet/lint-staged/issues/468)

Everyone should run `yarn install` on the root so the Husky hooks get installed too. Maybe announce it on the #dev channel.

In regards to the monitoring widgets, all info is here pavlovcik/uad-ui-archived#3

### CODE REVIEW

File: frontend/.eslintrc.js [hunk 1]
```diff
@@ -3,10 +3,14 @@
...
   extends: [
     "eslint:recommended",
     "plugin:@typescript-eslint/recommended",
     "prettier",
   ],
+  rules: {
+    "@typescript-eslint/explicit-module-boundary-types": "off",
+    "prettier/prettier": "error",
+  },
 };
```
Comment (0x4007) [inline frontend/.eslintrc.js]: I prefer to make these into JSONs!

File: frontend/components/context/connected.tsx [hunk 2]
```diff
@@ -19,72 +21,58 @@ export interface Balances {
...
+    manager,
+    setManager,
+  ] = useState<UbiquityAlgorithmicDollarManager | null>(null);
+  const [account, setAccount] = useState<EthAccount | null>(null);
+  const [balances, setBalances] = useState<Balances | null>(null);
+  const [twapPrice, setTwapPrice] = useState<BigNumber | null>(null);
+  const [contracts, setContracts] = useState<Contracts | null>(null);
 
-  const value = {
+  const value: IConnectedContext = {
```
Comment (0x4007) [inline frontend/components/context/connected.tsx]: I don't believe in the `I` prefix for interfaces, would be great to change these to `ConnectedContext`

File: frontend/next.config.js [hunk 3]
```diff
@@ -1,5 +1,11 @@
+const WindiCSS = require("windicss-webpack-plugin").default;
+
 module.exports = {
-    future: {
-        webpack5: true,
-    },
-}
\ No newline at end of file
+  webpack: (config) => {
+    config.plugins.push(new WindiCSS());
```
Comment (0x4007) [inline frontend/next.config.js]: You know my thoughts on this WindiCSS thing...

File: frontend/pages/_app.tsx [hunk 4]
```diff
@@ -1,21 +1,15 @@
 /* eslint-disable @typescript-eslint/explicit-module-boundary-types */
 import "./styles/index.css";
 import "./styles/proxima.css";
+import "windi.css";
```
Comment (0x4007) [inline frontend/pages/_app.tsx]: 🤨 ser pls

File: frontend/tsconfig.json [hunk 5]
```diff
@@ -20,6 +20,7 @@
   },
   "include": [
     "next-env.d.ts",
+    "windi-attributify.d.ts",
```
Comment (0x4007) [inline frontend/tsconfig.json]: ser pls

File: frontend/windi.config.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  attributify: true,
+  shortcuts: {},
+  theme: {
+    extend: {
+      letterSpacing: {
+        widest: ".2em",
+      },
+    },
+  },
+});
```
Comment (0x4007) [inline frontend/windi.config.ts]: pls

**Review by 0x4007 — CHANGES_REQUESTED** [review]

**Comment by Zequez** [comment]
@pavlovcik are you wanting me to remove WindiCSS? Because I would rather it be kept 😆

**Comment by 0x4007** [comment]
Yea @zgorizzo69 and I think it's irresponsible to introduce new technologies unless most of the industry is already using it.

**Comment by Zequez** [comment]
I mean, it's just CSS, is not like we're introducing groundbreaking tech; it's a CSS dev tool for our own dashboard. It's like using SASS, LESS, Webpack, (...or Bootstrap); it's not an architectural change or even a coding change; it's a tool to make CSS more maintainable long term; a set of CSS utilities that are modular, configurable and easy to use. [Did you explore the Tailwind website](https://tailwindcss.com/)? And [the article from the original author about the reasoning behind it](https://adamwathan.me/css-utility-classes-and-separation-of-concerns/)?

**Comment by Zequez** [comment]
There I added Tailwind in a way that we commit the style output to the repo; is that what you had in mind @pavlovcik ? I also changed WindiCSS with Tailwind just because Tailwind is more popular and known so people don't get confused.

**Comment by Zequez** [comment]
If this gets merged, then we can just close #2  because this branch includes that one; and if I merge back the last change to that branch then they are essentially the same branch. How the heck do people manage to make multiple PRs that change the same files without these being co-dependant? 😆

**Comment by 0x4007** [comment]
> There I added Tailwind in a way that we commit the style output to the repo; is that what you had in mind @pavlovcik ?

Yeah as long as it doesn't become a requirement for future developers to be productive then I think its fine. @zgorizzo69 any thoughts?

> If this gets merged, then we can just close #2 because this branch includes that one; and if I merge back the last change to that branch then they are essentially the same branch. How the heck do people manage to make multiple PRs that change the same files without these being co-dependant? 😆

I suppose you could branch off of a PR to do this if I am understanding correctly! Do what you think makes sense here. I believe you have the ability to close PRs as well right?

**Review by 0x4007 — APPROVED** [review]

**Comment by 0x4007** [comment]
I broke this @Zequez currently trying to figure out how to merge :(

File: frontend/components/common.tsx [hunk 7]
```diff
@@ -26,7 +26,7 @@ import UarRedeem from "./uar.redeem";
 import UbqBalance from "./ubq.balance";
 import { Contracts } from "../src/contracts";
 
-const PROD = process.env.NODE_ENV == "production";
+const PROD = true; //  process.env.NODE_ENV == "production";
```
Comment (0x4007) [inline frontend/components/common.tsx:29]: Whoops, this is nice for seeing the video in the background in dev though!

File: tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (0x4007) [inline tsconfig.json:1]: Moved this up one because my vscode was not picking up its rules in my text editor.

**Pull request merged**